File: gerenciapet-frontend/src/app/models/Address.ts
export default class Address {
	public street: string = "";
	public houseNumber: string = "";
	public addInfo: string = "";
	public neighborhood: string = "";
	public city: string = "";
	public state: string = "";
	public refPoint: string = "";
	public cep: string = "";

    constructor(object?: any) {
        if (object !== undefined) {
            this.updateAddress(object);
        }
    }

	public updateAddress(object : any){
		this.street = object.street?.toUpperCase() ?? "";
		this.houseNumber = object.houseNumber;
		this.addInfo = object.addInfo;
		this.neighborhood = object.neighborhood?.toUpperCase() ?? "";
        this.city = object.city?.toUpperCase() ?? "";
        this.state = object.state?.toUpperCase() ?? "";
		this.refPoint = object.refPoint;
		this.cep = object.cep;
	}

    public print(): void {
        console.log("street: " + this.street);
        console.log("houseNumber: " + this.houseNumber);
        console.log("addInfo: " + this.addInfo);
        console.log("neighborhood: " + this.neighborhood);
        console.log("city: " + this.city);
        console.log("state: " + this.state);
        console.log("refPoint: " + this.refPoint);
        console.log("cep: " + this.cep);
    }
}

File: gerenciapet-backend/src/main/java/br/ufes/gerenciapet/backend/model/Address.java
package br.ufes.gerenciapet.backend.model;

import javax.persistence.Embeddable;

@Embeddable
public class Address {

    private String street;

    private String houseNumber;

    private String addInfo; //Complemento

    private String neighborhood;

    private String city;

    private String state;

    private String refPoint;

    private String cep; // Não sabe ao certo se vai utilizar

    // #################################################################################################################
    // Getters and Setters
    // #################################################################################################################

    public String getStreet() {
        return street;
    }

    public void setStreet(String street) {
        this.street = street;
    }

    public String getHouseNumber() {
        return houseNumber;
    }

    public void setHouseNumber(String houseNumber) {
        this.houseNumber = houseNumber;
    }

    public String getCep() {
        return cep;
    }

    public void setCep(String cep) {
        this.cep = cep;
    }

    public String getNeighborhood() {
        return neighborhood;
    }

    public void setNeighborhood(String neighborhood) {
        this.neighborhood = neighborhood;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getAddInfo() { return addInfo; }

    public void setAddInfo(String addInfo) { this.addInfo = addInfo; }

    public String getRefPoint() { return refPoint; }

    public void setRefPoint(String refPoint) { this.refPoint = refPoint; }
}
